admin/registration_providers/forms.py

```diff
@@ -0,0 +1,72 @@
+import bleach
+
+from django import forms
+
+from osf.models import RegistrationProvider, Subject
+from admin.base.utils import get_nodelicense_choices, get_toplevel_subjects, get_defaultlicense_choices
+
+
+class RegistrationProviderForm(forms.ModelForm):
+    toplevel_subjects = forms.MultipleChoiceField(widget=forms.CheckboxSelectMultiple(), required=False)
+    subjects_chosen = forms.CharField(widget=forms.HiddenInput(), required=False)
```

Comment:
Non-custom taxonomy fields (and their relics throughout the view / template / static files) should be removed. Tested locally, and setting these didn't do anything.